Sources/HTTP/Models/Client/HTTP+Client.swift

```diff
@@ -1,79 +1,57 @@
 import Transport
+import Sockets
 
 public enum ClientError: Swift.Error {
     case invalidRequestHost
     case invalidRequestScheme
     case invalidRequestPort
     case unableToConnect
     case userInfoNotAllowedOnHTTP
+    case missingHost
 }
 
-public typealias BasicClient = Client<TCPClientStream, Serializer<Request>, Parser<Response>>
-
-public final class Client<
-    ClientStreamType: ClientStream,
-    SerializerType: TransferSerializer,
-    ParserType: TransferParser
->: ClientProtocol
-where
-    ParserType.MessageType == Response,
-    SerializerType.MessageType == Request
+public typealias TCPClient = BasicClient<
+    TCPInternetSocket,
+    Serializer<Request, StreamBuffer<TCPInternetSocket>>,
+    Parser<Response, StreamBuffer<TCPInternetSocket>>
+>
+
+import TLS
+
+public typealias TLSTCPClient = BasicClient<
```

Comment:
What was the motivation between this and the client name change? I preferred old style, but if there's good reason, worth a change